assets/scripts/get-actions.js

```diff
@@ -1,12 +1,116 @@
 import { ucfirst } from './ucfirst';
 
 /**
- * Crate attributes actions from blocks manifest.json.
+ * This method is for atributes with property type `object` with any number of values, eg. `content`, `url`, `type` etc.
+ * This function generates callback for each of the values from attribute. Generate callback with name onChange${attribute_name}${propery_name}
+ * Example `onChangeHeadingColor` where atribute name is `heading` and object property is `color`
+ *
+ * @param {object} setAttributes Method for saving atributes.
+ * @param {object} attributes All atributes from namifest.json
+ * @param {string} key Came of the property in manifes.
+ * @param {object} propsAttributes Current atributes when this function executes.
+ *
+ * @since 1.0.5 Ucfirst path hotfix.
+ * @since 1.0.4 Moving to scripts folder.
+ * @since 1.0.0
+ */
+
+const multiplePropsActions = (setAttributes, attributes, key, propsAttributes) => {
+  const output = {};
+
+  // Set output as a object key with anonimus function callback.
+  for (const propType in attributes[key]) {
+
+    // Create functions for default values.
+    if (propType === 'default') {
+
+      for (const propName in attributes[key][propType]) {
+        if (attributes[key][propType].hasOwnProperty(propName)) {
+
+          output[`onChange${ucfirst(key)}${ucfirst(propName)}`] = function(value) {
+            setAttributes({
+              [key]: {
+                ...propsAttributes[key],
+                [propName]: value,
+              },
+            });
+          };
+
+        }
+      }
+    }
+  }
+
+  return output;
+};
+
+/**
+ * This method is for atributes with single property.
+ * This function generates callback for that propertiy value.
+ *
+ * @param {object} setAttributes Method for saving atributes.
+ * @param {string} key Came of the property in manifes.
+ *
+ * @since 1.0.5 Ucfirst path hotfix.
+ * @since 1.0.4 Moving to scripts folder.
+ * @since 1.0.0
+ */
+
+const siglePropsAction = (setAttributes, key) => {
+  const output = {};
+
+  // Set output as a object key with anonimus function callback.
+  // Keys first name must be uppercased.
+  output[`onChange${ucfirst(key)}`] = function(value) {
+    setAttributes({
+      [key]: value,
+    });
+  };
+
+  return output;
+};
+
+
+/**
+ * This method is for media atributes, it is property type `object` with default values of `id`, `url`, `title`.
```

Comment:
```suggestion
 * This method is used for setting media attributes.
 * Fill in the rest of the documentation.
```

Please read the documentation when you are writing it. It needs to be clear what the method does when you read the documentation, otherwise we are doing this in vain...

Reply:
👍